crates/faucet/src/lib.rs

```diff
@@ -187,10 +186,12 @@ impl Faucet {
     pub async fn run(
         mut self,
         mut requests: Receiver<(MintRequest, MintResponseSender)>,
+        batch_size: usize,
     ) -> anyhow::Result<()> {
         let mut buffer = Vec::new();
 
-        while requests.recv_many(&mut buffer, BATCH_SIZE).await > 0 {
+        // TODO: batching 10 or more requests fails, see https://github.com/0xMiden/miden-faucet/issues/85
+        while requests.recv_many(&mut buffer, batch_size).await > 0 {
```

Comment:
The issue should now be fixed - though, would be good to confirm this. If all looks good, we should remove the `TODO` and probably increase the default to something like 32 (or maybe even more) notes per tx.